### ISSUE TITLE
Cropper does not work properly in components with ChangeDetectionStrategy.OnPush

### ISSUE THREAD

**Issue by blazekv** [vugar005/ngx-awesome-uploader#44]
**Description:**
Cropper overlay will not close if ngx-awesome-uploader is used inside component with 
`changeDetection: ChangeDetectionStrategy.OnPush`. Modal window is still visible even after Crop click. On next click error will raise: `Cannot read property 'getCroppedCanvas' of undefined`

**How to reproduce:**
Just change changeDetection strategy on component where ngx-awesome-uploader is used. For example on demo:

```
import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { DemoFilePickerAdapter } from '../demo-file-picker.adapter';
import { HttpClient } from '@angular/common/http';
@Component({
  selector: 'simple-demo',
  templateUrl: './simple-demo.component.html',
  styleUrls: ['./simple-demo.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class SimpleDemoComponent implements OnInit {
  adapter = new DemoFilePickerAdapter(this.http);

  constructor(private http: HttpClient) { }

  ngOnInit() {
  }

}

```

**Workaround**
1. inject ChangeDetectorRef
2. get reference to FilePickerComponent (setter is needed only in case when filepicker is not accessible at AfterViewInit method)
3. subscribe to cropClosed action and mark for check

```
import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { DemoFilePickerAdapter } from '../demo-file-picker.adapter';
import { HttpClient } from '@angular/common/http';
@Component({
  selector: 'simple-demo',
  templateUrl: './simple-demo.component.html',
  styleUrls: ['./simple-demo.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class SimpleDemoComponent implements OnInit {
  adapter = new DemoFilePickerAdapter(this.http);

  constructor(private http: HttpClient, private cd: ChangeDetectorRef) { }

  ngOnInit() {
  }

  @ViewChild(FilePickerComponent)
  set filePickerComponent(filePickerComponent) {
    filePickerComponent.cropClosed$.subscribe(value => {
      this.cd.markForCheck();
    });
  }

}
```

**Comment by vugar005** [comment]
Hi @blazekv thanks for raising issue. Sorry for later reply. Yes it is going to be fixed on next version. Thanks!

**Comment by vugar005** [comment]
@blazekv  Thanks for raising issue. Fixed from version 12.0.1.
